src/main/java/meteordevelopment/meteorclient/systems/modules/combat/AutoWeapon.java

```diff
@@ -42,14 +43,84 @@ public class AutoWeapon extends Module {
         .build()
     );
 
+    private final Setting<Boolean> switchSecondary = sgGeneral.add(new BoolSetting.Builder()
+        .name("switch to secondary weapon on one-hit")
+        .description("Switches to secondary weapon when the victim's health goes below the desired number")
+        .defaultValue(false)
+        .build()
+    );
+    private final Setting<Weapon2> weapon2 = sgGeneral.add(new EnumSetting.Builder<Weapon2>()
+        .name("secondary weapon")
+        .description("what weapon to switch to on the desired value")
+        .defaultValue(Weapon2.Sword)
+        .visible(switchSecondary::get)
+        .build()
+    );
+    private final Setting<Double> healthThreshold = sgGeneral.add(new DoubleSetting.Builder()
+        .name("health threshold")
+        .description("When to trigger the secondary weapon based on victim's health")
+        .defaultValue(5)
+        .sliderRange(1, 36)
+        .visible(switchSecondary::get)
+        .build()
+    );
+
+    public int bsfr(LivingEntity entity) {
+        return (int) entity.getHealth();
+    }
+
     public AutoWeapon() {
         super(Categories.Combat, "auto-weapon", "Finds the best weapon to use in your hotbar.");
     }
 
     @EventHandler
     private void onAttack(AttackEntityEvent event) {
+        if (switchSecondary.get() && bsfr((LivingEntity) event.entity) < healthThreshold.get()) {
+            InvUtils.swap(getSecondaryWeapon(EntityUtils.getGroup(event.entity), (int) (bsfr((LivingEntity) event.entity))), false);
+        } else
```

Comment:
the else case should be on a newline and should also have brackets around

if (...) {
    ...
}
else {
    ...
}